docs: add v3 migration guide (#1806)
siemens/ix: packages/documentation/docs/home/migration/3_0_0/index.md

sidebar_title: Updating to v3
title: Updating to v3
---

# Updating to v3.0.0

Welcome to the migration guide for upgrading from Siemens Industrial Experience design system v2 to v3. Find a list of all breaking changes [here](https://github.com/siemens/ix/blob/main/BREAKING_CHANGES.md#v300).

## Removed light and foundation Figma libraries

The following Figma libraries have been replaced by the _iX Component_ library:

- iX Component - Brand Light
- iX Foundation - Brand Light
- iX Foundation - Brand Dark
- iX Helpers

Follow the instructions below to update your Figma files to the new iX Component library:

1. On the left panel, click **Assets**.
2. Click on the <ix-icon name="book" size="16"></ix-icon> icon to open the libraries dialog.
3. Select a library you want to replace.
4. Click **Swap library**.
5. Choose **iX Components**.
6. Confirm and repeat for the other libraries.

Figma libraries for v2.0.0 are in the **Archive** project (find the link in the main library).

:::info
We have also prepared a video showing the process in our internal Siemens learning world [here](https://mylearningworld.siemens.com/web/es/app/knowledge-board/lex_auth_0142543970762915841020).
:::

> ## Removed package `@siemens/ix-icons` from `@siemens/ix` base library

Reviewer: ```suggestion
## Changed loading of icons from package `@siemens/ix-icons`
```

```

### Changed usage of icons in pure HTML

```html
<ix-icon name="star"></ix-icon>
```

### Custom asset path for icons

In order to be able to load custom icons, we need to configure the domain of the asset path. This can either be done via `meta`-tag or with the `setAssetPath`-function.

#### With `meta`-tag

```html
<html>
  <head>
    <!-- Some other tags -->
    <meta name="ix-icons:path" content="https://some-resource-domain" />
  </head>
  <body></body>
</html>
```

```html
<IxIcon name="star"></IxIcon>
```

Above code will fetch the SVG from `https://some-resource-domain/star.svg`.

#### With `setAssetPath`-function

Make sure to call the `setAssetPath`-function before using the `IxIcon` component (e.g. in the main file).

```ts
import { setAssetPath } from '@siemens/ix-icons/components';

setAssetPath('https://some-resource-domain');
```

```html
<IxIcon name="star"></IxIcon>
```

Above code will fetch the SVG from `https://some-resource-domain/star.svg`

This will preload all icons without including the SVGs as assets, which results in a larger bundle size.
Therefore, this approach is **NOT recommended**.

## Changed props to `@internal`

> The following component members marked as internal by comment have been changed to annotated as `@internal`:

Reviewer: ```suggestion
The following component members previously marked as internal by comment have been changed to annotated as `@internal`:
```

- The class `.table-striped` is now `.ix-table-striped`.

Please note that they are not based on Bootstrap anymore. If your tables rely on additional Bootstrap functionality (e.g. hoverable rows), you have to install Bootstrap on your own.

## Added namespacing of standard HTML elements

We decided to add the prefix `ix-` to our global styles of standard HTML elements like `input`, `textarea` and `label`.

This ensures that you can use other CSS libraries alongside iX to prevent class collisions. To ensure that your styles are not broken after you update to v3, you have to provide the class `ix-form-control` to `input` and `textarea`.

The standard `label` element needs the `ix-form-label` class.

Example:

Change

```html
<input type="text" /> <input type="text" class="form-control" />
```

to

```html
<input type="text" class="ix-form-control" /> <input type="text" class="ix-form-control" />
```

## Global style updates

### Changed `input` width value

We changed the default width of the `input` element from `width: 100%` to `width: auto`

## Component updates

#### ix-action-card

- Replaced `insight` and `notification` variants with `outline` and `filled`.

#### ix-card

> - Replaced `insight` and `notification` variants have been with `outline` and `filled`.

Reviewer: ```suggestion
- Replaced `insight` and `notification` variants with `outline` and `filled`.
```

#### ix-chip

- Replaced `color` property with `chipColor`.

#### ix-date-picker

- Removed `individual` and `eventDelimiter` attributes.
- Replaced `textSelectedDate` property with `i18nDone`.
- Replaced `done` event with `dateSelect`.

#### ix-datetime-picker

- Replaced `textSelectedDate` property with `i18nDone`.
- Replaced `done` event replaced with `dateSelect`.
- Removed `eventDelimiter` property.

#### ix-event-list

- Replaced `color` attribute with `itemColor`.

#### ix-icon-button

- Replaced `color` attribute with `iconColor`.

> - Removed size `32`.

Reviewer: ```suggestion
- Removed option `32` from attribute `size.
```

#### ix-menu

- Removed `maxVisibleMenuItems` attribute.

#### ix-menu-item

- Replaced `tabIcon` attribute with `icon`.

#### ix-modal

- Replaced `keyboard` attribute with `closeOnEscape`.

#### ix-pill

- Replaced `color` attribute with `pillColor`.

#### ix-push-card

- Replaced `insight` and `notification` variants with `outline` and `filled`.

#### ix-select

- Replaced `selectedIndices` attribute with `value`.
- Replaced `itemSelectionChange` event with `valueC**hange`.

#### ix-select-item

> - Replaced `value` attribute type with `string`.

Reviewer: ```suggestion
- Changed type of attribute `value` to `string`.
```
